### ISSUE TITLE
refactor CI bazel options

### ISSUE THREAD

**Issue by lizan** [envoyproxy/envoy#5730]
It is extremely hard to follow bazel options being set via environment variables in several places across different scripts. I'm proposing having a single `.bazelrc.ci` with several config to control CI bazel options. We can still leave *_EXTRA_OPTIONS for customized run though.

If no one object I will work on it.

**Comment by mattklein123** [comment]
+1 I agree what we have now is very confusing.

**Comment by lizan** [comment]
keep alive, will work on this after clearing up dependencies from prebuilt.
